From a8cecac6bd393f3c9ba2f6f7333e99e7f7fcee2c Mon Sep 17 00:00:00 2001
From: Eugene Susla <eugenesusla@google.com>
Date: Mon, 26 Oct 2020 09:08:49 -0700
Subject: [PATCH 25/73] [Q_asb_2021-04] RESTRICT AUTOMERGE Prevent non-system
 overlays from showing over CDM UI

Since CDM grants privileges, it should have the same overlay
policy as permission UI

Test: use an app wit ha visible overlay to ensure
the overlay disappears when CDM is shown
Fixes: 171221090

Change-Id: I4daaee7d8b710a72f6166cbb2252ef8af84c2c60
(cherry picked from commit 2d5a635fec95667d96ba9dd1007d393299018cb2)
---
 .../android/companiondevicemanager/DeviceChooserActivity.java  | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/packages/CompanionDeviceManager/src/com/android/companiondevicemanager/DeviceChooserActivity.java b/packages/CompanionDeviceManager/src/com/android/companiondevicemanager/DeviceChooserActivity.java
index 16ef59f201f1..7bc9993751d4 100644
--- a/packages/CompanionDeviceManager/src/com/android/companiondevicemanager/DeviceChooserActivity.java
+++ b/packages/CompanionDeviceManager/src/com/android/companiondevicemanager/DeviceChooserActivity.java
@@ -17,6 +17,7 @@
 package com.android.companiondevicemanager;
 
 import static android.companion.BluetoothDeviceFilterUtils.getDeviceMacAddress;
+import static android.view.WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS;
 
 import android.app.Activity;
 import android.companion.CompanionDeviceManager;
@@ -56,6 +57,8 @@ public class DeviceChooserActivity extends Activity {
             Log.e(LOG_TAG, "About to show UI, but no devices to show");
         }
 
+        getWindow().addSystemFlags(SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
+
         if (getService().mRequest.isSingleDevice()) {
             setContentView(R.layout.device_confirmation);
             final DeviceFilterPair selectedDevice = getService().mDevicesFound.get(0);
-- 
2.35.1

